src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,138 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+
+    private T[] elements;
+    private int size = 0;
+
+    public ArrayList() {
+        elements = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
+    private void ensureCapacity() {
+        T[] newElements = (T[]) new Object[elements.length * 2];
+        System.arraycopy(elements, 0, newElements, 0, size);
+        elements = newElements;
+    }
```

Comment:
The method `ensureCapacity` is not following the Single Responsibility principle. It is doing two things: creating a new array and copying the old array into the new one. It would be better to divide this method into two separate methods: `createNewArray` and `copyOldArrayIntoNewOne`.